nameable.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+# Nameable is a base class for objects that have a "name" attribute and need to be
+# able to produce a "correct name" based on that attribute.
+class Nameable
+  def correct_name
+    raise NotImplementedError, 'Subclass must implement abstract method'
```

Comment:
- [ Optional ] I would suggest you to add the name of the class and the method, it will make your code look more professional
` raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"`